### ISSUE TITLE
Build no longer works

### ISSUE THREAD

**Issue by Snickdx** [gdg-x/hoverboard#1404]
**Describe the bug**
The build command doesn't seem to output the HTML files of the site and its copying my service account into the build directory.

I wish I could do more than just complain but I just don't have enough experience with polymer and rollup. At this point, I'm just curious to know if anyone has gotten this project to work recently.

PS: This project now requires a paid firebase plan in order to run firestore:init

**To Reproduce**
Add firebase config
Add service account
npm install
npm run build

**Expected behavior**
Build site to out-tsc directory (was previously called build)

**Screenshots**
![image](https://user-images.githubusercontent.com/5845054/95480759-45577c00-095a-11eb-9e57-6546795b550d.png)

**Desktop (please complete the following information):**

- OS: Windows/Linux
- Browser Chrome

**Additional context**
Hi I have been using this project since 2018 and I really like it but today It seems no longer feasible for my chapter to use. The install command takes ~20mins. I have been trying for weeks to set up and I'm wondering if there's anyone who can actually pull it off with the way it is now.

**Comment by abraham** [comment]
How recent a version of Hoverboard are you using? Have you tried on the latest version?

`npm run build` should not result in a `build` directory. It should create a `dist` directory with the production ready version of the app. There will be an `out-tsc` as well but this is an intermediate step that you should not have to interact with at all.

What is the complete command output you get when you run `npm run build`?

Use `npm ci` instead of `npm install`. It's newer and faster. When I run `npm ci` it takes about 31 seconds. I do have a fast connection and computer but that should not cause the difference between 31 seconds and 20 minutes.

What are the computer system specs and internet speed you have?

**Comment by zptoth** [comment]
The below method works for me. I use it for every single deployment, even though step 1 and 2 should not be performed every time.

```
npx firebase firestore:delete --recursive --all-collections
npm run firestore:init
npm run deploy

```

Optionally if you use multiple firebase projects it's a good practice to execute **npx firebase use <project-id>** before the commands above.

**Comment by zptoth** [comment]
For local testing I use **npm run build; npm start** but not all modifications could be tested this way. Some require a full deployment to firebase.

**Comment by zptoth** [comment]
I have even developed a small (not too professional, but useful) deployment script that handles **development** and **production** versions of the site that could be deployed from two separate (dev and prod) branches from the same repo. I can share it with you if you are interested.

**Comment by abraham** [comment]
You shouldn't need to `npm run build` before `npm start`. `npm start` ignores most of the production build that `npm run build` creates.

There is working being done on a better local development experience in https://github.com/gdg-x/hoverboard/pull/1316. That combined with https://github.com/gdg-x/hoverboard/pull/1384 and https://github.com/gdg-x/hoverboard/issues/698 I'm hoping there will be a pretty good experience for developers.

**Comment by zptoth** [comment]
You got me with the `npm run build` part but for the other one I couldn't find any better solution but doing a delete and a new init all the time. This makes it possible for me to do changes locally in the code and deploy them by executing one single script.

**Comment by Snickdx** [comment]
Hi an update: I tried from fresh using a cloud environment called gitpod with npm ci and I got the build to work! Will use that from now on instead of windows on my machine. 

One thing I've also noticed is that any data on the site will not load if any part the firebase-default-data.json is malformed. Eg speaker ids not in the speaker session, miss-spelling/case of properties. It would be nice if there was some kind of runtime error that explains this to the developer.
